addons/html_editor/static/tests/link/popover.test.js

```diff
@@ -757,3 +760,94 @@ describe("link preview", () => {
+        click("img");
+        await waitFor(".o-we-toolbar");
+        expect("button[name='image_unlink']").toHaveCount(1);
+        click("button[name='image_unlink']");
+        await animationFrame();
+        expect(cleanLinkArtifacts(getContent(el))).toBe(
+            `<p>ab<a href="#">cd[</a><img src="${base64Img}"><a href="#">]ef</a>g</p>`
+        );
+        expect(".o-we-linkpopover").toHaveCount(0);
+    });
+    test("can remove link from a selection of an inline image + text", async () => {
+        const { el } = await setupEditor(
+            `<p>ab<a href="#">c[d<img src="${base64Img}">e]f</a>g</p>`
+        );
+        await waitFor(".o-we-toolbar");
+        click(".o-we-toolbar .fa-unlink");
+        await tick();
+        expect(cleanLinkArtifacts(getContent(el))).toBe(
+            `<p>ab<a href="#">c</a>[d<img src="${base64Img}">e]<a href="#">f</a>g</p>`
+        );
+    });
+    test("can remove link from a selection (ltr) with multiple inline images", async () => {
+        const { el } = await setupEditor(
+            `<p>ab<a href="#">c[d<img src="${base64Img}">e<img src="${base64Img}">f]g</a>h</p>`
+        );
+        await waitFor(".o-we-toolbar");
+        click(".o-we-toolbar .fa-unlink");
+        await tick();
+        expect(cleanLinkArtifacts(getContent(el))).toBe(
+            `<p>ab<a href="#">c</a>[d<img src="${base64Img}">e<img src="${base64Img}">f]<a href="#">g</a>h</p>`
+        );
+    });
+    test("can remove link from a selection (rtl) with multiple inline images", async () => {
+        const { el } = await setupEditor(
+            `<p>ab<a href="#">c]d<img src="${base64Img}">e<img src="${base64Img}">f[g</a>h</p>`
+        );
+        await waitFor(".o-we-toolbar");
+        click(".o-we-toolbar .fa-unlink");
+        await tick();
+        expect(cleanLinkArtifacts(getContent(el))).toBe(
+            `<p>ab<a href="#">c</a>]d<img src="${base64Img}">e<img src="${base64Img}">f[<a href="#">g</a>h</p>`
+        );
+    });
+    test("can remove link from a selection (ltr) with multiple inline images acrossing different links", async () => {
+        const { el } = await setupEditor(
+            `<p>ab<a href="#">c[d<img src="${base64Img}">e</a>xx<a href="#">f<img src="${base64Img}">g]h</a>i</p>`
+        );
+        await waitFor(".o-we-toolbar");
+        click(".o-we-toolbar .fa-unlink");
+        await tick();
+        expect(cleanLinkArtifacts(getContent(el))).toBe(
+            `<p>ab<a href="#">c</a>[d<img src="${base64Img}">exxf<img src="${base64Img}">g]<a href="#">h</a>i</p>`
+        );
+    });
+    test("can remove link from a selection (rtl) with multiple inline images acrossing different links", async () => {
+        const { el } = await setupEditor(
+            `<p>ab<a href="#">c]d<img src="${base64Img}">e</a>xx<a href="#">f<img src="${base64Img}">g[h</a>i</p>`
+        );
+        await waitFor(".o-we-toolbar");
+        click(".o-we-toolbar .fa-unlink");
+        await tick();
```

Comment:
why we need this tick ?

Reply:
I used it to make sure the unlinking is executed before the comparing. I worry it can fail the expect in case it might take longer